mirdata/irmas.py

```diff
@@ -0,0 +1,406 @@
+                                        irmas_dict[irmas_id_nod_no_wav] = {
+                                            'genre': genre_code,
+                                            'drum': None,
+                                            'train': True,
+                                        }
+            if 'Test' in directory:
+                for root_, dirs_, files_ in os.walk(os.path.join(data_home, directory)):
+                    for directory_ in dirs_:
+                        for root__, dirs__, files__ in os.walk(
+                            os.path.join(data_home, directory, directory_)
+                        ):
+                            for file in files__:
+                                if file.endswith('.wav'):
+                                    test_index = str(count)
+                                    irmas_dict[test_index] = {
+                                        'genre': None,
+                                        'drum': None,
+                                        'train': False,
+                                    }
+                                    count += 1
+
+    irmas_dict["data_home"] = data_home
+
+    return irmas_dict
+
+
+DATA = utils.LargeData(
+    'irmas_index.json', _load_metadata
+)
+
+
+class Track(track.Track):
+    """IRMAS track class
+
+    Args:
+        track_id (str): track id of the track
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets/Mridangam-Stroke`
+
+    Attributes:
+        track_id (str): track id
+        train (bool): flag to identify if the track is from the training of the testing dataset
+        genre (str): string containing the namecode of the genre of the track.
+        drum (bool): flag to identify if the track contains drums or not.
+    """
+
+    def __init__(self, track_id, data_home=None):
+        if track_id not in DATA.index:
+            raise ValueError('{} is not a valid track ID in Example'.format(track_id))
+
+        self.track_id = track_id
+
+        if data_home is None:
+            data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+        self._data_home = data_home
+
+        self._track_paths = DATA.index[track_id]
+        self.audio_path = os.path.join(self._data_home, self._track_paths['audio'][0])
+
+        metadata = DATA.metadata(data_home)
+        if metadata is not None and track_id in metadata:
+            self._track_metadata = metadata[track_id]
+        else:
+            self._track_metadata = {'genre': None, 'drum': None, 'train': None}
+
+        self.annotation_path = os.path.join(
+            self._data_home, self._track_paths['annotation'][0]
+        )
+        self.genre = self._track_metadata['genre']
+        self.drum = self._track_metadata['drum']
+        self.train = self._track_metadata['train']
+
+    @utils.cached_property
+    def predominant_instrument(self):
```

Comment:
If there's only one instrument per recording, I'll name this `instrument` to be consistent with other datasets

Reply:
In fact, the purpose of this dataset is to detect the predominant instrument(s) in a track. In many tracks, most of them (especially in the testing set), the tracks have many instruments and the idea is to detect the most prominent one. So should I leave it this way or should I call name it `instrument` and then specify in the documentation that it is referring to the predominant one?

Reply:
Mmmm... I see. So you have many instruments per track, but you're interested in finding the predominant one, right? Would it make sense to have **both** `predominant_instruments` **and** `instruments`? In the first one, you have the target instrument, and in the second, you have all instruments in the track (for e.g. instrument recognition). Just an idea, don't know if it makes sense for this dataset, maybe @nkundiushuti has comments on this?

Reply:
The problem here is that there is no annotation of the complete instrument line-up of the tracks. For both training and testing samples, we just have the predominant instrument, without having any information about the rest of the  background instruments.

Reply:
Ooh I see, thanks. It seems this case is different from the datasets we already have with instruments (`tinysol`, `medley_db_solos` and `medley_db_pitch`). @rabitt should we go for `instrument` to be compatible with those even though there are multiple instruments in the mix, but we "care" only about one?